pkg/crypto/keys/publickey.go

```diff
@@ -34,6 +34,37 @@ func (keys PublicKeys) Less(i, j int) bool {
 	return keys[i].Y.Cmp(keys[j].Y) == -1
 }
 
+// DecodeBytes decodes a PublicKeys from the given slice of bytes.
+func (keys PublicKeys) DecodeBytes(data []byte) error {
+	b := io.NewBinReaderFromBuf(data)
+	b.ReadArray(keys)
+	return b.Err
+}
+
+// Contains checks whether passed param contained in PublicKeys.
+func (keys PublicKeys) Contains(pKey *PublicKey) bool {
+	for _, key := range keys {
+		if key == pKey {
+			return true
+		}
+	}
+	return false
+}
+
+// Unique returns set of public keys.
+func (keys PublicKeys) Unique() PublicKeys {
+	unique := make([]*PublicKey, 0, keys.Len())
+	pkMap := make(map[*PublicKey]bool)
```

Comment:
It is probably better not to use pointers here, because `PublicKey`s can be equivalent even if the pointers are not the same (e.g. they both are result of a parsing).